npm/test/filecheckprocessing.test.ts

```diff
@@ -0,0 +1,140 @@
+const mockGraphqlClient = {
+  updateProgressBar: jest.fn(),
+  getConsignmentData: jest.fn(),
+  getConsignmentId: jest.fn()
+}
+import {
+  IFileCheckProcessed,
+  updateProgressBar,
+  getConsignmentData,
+  getConsignmentId
+} from "../src/filechecks/file-check-processing"
+import {
+  GetFileCheckProgressQueryVariables,
+  GetFileCheckProgressQuery
+} from "@nationalarchives/tdr-generated-graphql"
+import { GraphqlClient } from "../src/graphql"
+import { mockKeycloakInstance } from "./utils"
+import { DocumentNode } from "graphql"
+import { FetchResult } from "apollo-boost"
+
+jest.mock("../src/graphql")
+
+class GraphqlClientSuccess {
+  mutation: (
+    query: DocumentNode,
+    variables: GetFileCheckProgressQueryVariables
+  ) => Promise<FetchResult<GetFileCheckProgressQuery>> = async (_, __) => {
+    const data: GetFileCheckProgressQuery = {
+      getConsignment: {
+        totalFiles: 10,
+        fileChecks: {
+          antivirusProgress: { filesProcessed: 2 },
+          checksumProgress: { filesProcessed: 3 }
+        }
+      }
+    }
+    return { data }
+  }
+}
+
+class GraphqlClientEmptyData {
+  mutation: (
+    query: DocumentNode,
+    variables: GetFileCheckProgressQueryVariables
+  ) => Promise<FetchResult<GetFileCheckProgressQuery>> = async (_, __) => {
+    const data: GetFileCheckProgressQuery = {
+      getConsignment: null
+    }
+    return { data }
+  }
+}
+
+class GraphqlClientFailure {
+  mutation: (
+    query: DocumentNode,
+    variables: GetFileCheckProgressQueryVariables
+  ) => Promise<FetchResult<GetFileCheckProgressQuery>> = (_, __) => {
+    return Promise.reject("error")
+  }
+}
+
+test.each`
+  processed | total   | expected
```

Comment:
This is really cool! Much easier to see all the expected/controlled data :open_mouth: